src/VisualStudio/CSharp/Impl/Options/AutomationObject.cs

```diff
@@ -739,7 +743,8 @@ public int Wrapping_IgnoreSpacesAroundBinaryOperators
 
             set
             {
-                _workspace.Options = _workspace.Options.WithChangedOption(CSharpFormattingOptions.SpacingAroundBinaryOperator, value);
+                _workspace.TryApplyChanges(_workspace.CurrentSolution.WithOptions(_workspace.Options
```

Comment:
💭 Shouldn't we be asserting that each of these calls returns true?

Reply:
Did you mean for the calls in this file or the entire PR? All the added API calls to TryApplyChanges are based off CurrentSolution.WithOptions, so they should ideally always succeed. If not, it would be better to add tests to guard those cases.

Reply:
Should we perhaps add `ApplyChanges` API that throws?

---
In reply to: [366658785](https://github.com/dotnet/roslyn/pull/40934#discussion_r366658785) [](ancestors = 366658785)